sharding/interfaces.go

```diff
@@ -1,20 +1,62 @@
 package sharding
 
-// Service defines items that can be registered into a sharding node's serviceFuncs.
-//
-// life-cycle management is delegated to the sharding node. The service is allowed to
-// initialize itself upon creation, but no goroutines should be spun up outside of the
-// Start method.
-type Service interface {
-	// Start is called after all services have been constructed and the networking
-	// layer was also initialized to spawn any goroutines required by the service.
+import (
+	"fmt"
+	"reflect"
+)
+
+// Node defines a a sharding-enabled Ethereum instance that provides
+// full control and shared access of necessary components and services
+// for a sharded Ethereum blockchain.
+type Node interface {
 	Start() error
+	Close() error
+	Register(constructor ServiceConstructor) error
+}
+
+// ShardP2P defines an interface for a peer-to-peer service in a
+// sharded Ethereum blockchain.
+type ShardP2P interface{}
 
-	// Stop terminates all goroutines belonging to the service, blocking until they
-	// are all terminated.
+// TXPool defines an interface for a transaction pool service that handles
+// incoming shard transactions in the network.
+type TXPool interface{}
+
+// Actor refers to either a notary, proposer, or observer in the sharding spec.
+type Actor interface {
```

Comment:
Do you expect that the actor interface will be anything more than a service?

If so, please use interface composition
```golang
type Actor interface {
   Service
   // TODO: Add actor specific methods.
}
````

Otherwise just refer to the actor as a service interface.